interfaces/connection.go

```diff
@@ -103,19 +103,31 @@ func getAttribute(snapName string, ifaceName string, staticAttrs map[string]inte
 }
 
 // NewConnectedSlot creates an object representing a connected slot.
-func NewConnectedSlot(slot *snap.SlotInfo, dynamicAttrs map[string]interface{}) *ConnectedSlot {
+func NewConnectedSlot(slot *snap.SlotInfo, staticAttrs, dynamicAttrs map[string]interface{}) *ConnectedSlot {
```

Comment:
Wondering if this could be:
```
type EndpointAttributes {
    Static map[string]interface{}
    Dynamic map[string]interface{}
}
func NewConnectedSlot(slot *snap.SlotInfo attrs *EndpointAttributes) *ConnectedSlot {
   ...
}
func NewConnectedPlug(slot *snap.PlugInfo attrs *EndpointAttributes) *ConnectedPlug {
   ...
}
```

Reply:
Yes, this is a nice idea indeed, thanks! But I think I'd rather keep it for later if we want to pursue it, the changes would be even bigger then they are now.